astroid/brain/brain_typing.py

```diff
@@ -134,7 +134,14 @@ def infer_typing_typevar_or_newtype(node, context_itton=None):
         raise UseInferenceDefault
 
     typename = node.args[0].as_string().strip("'")
-    node = extract_node(TYPING_TYPE_TEMPLATE.format(typename))
+    node = ClassDef(
```

Comment:
Thanks for such quick feedback👍.
I started a short vacation right after opening this so I’ll think about a test case when I can.

Reply:
Hey @DanielNoord I've taken a look finally :)

It seems to me there is no need for `__getitem__` to be attached to the astroid node representing either `typing.TypeVar` or `typing.NewType` since neither of these are subscriptable. What do you think?
Similar [point](https://github.com/python/cpython/blob/3.11/Lib/typing.py#L1323C28-L1323C28) for `__args__` since that is used to fetch the arguments used in subscripting. 

As a sidenote - `__getitem__` is already part of `nodes.ClassDef` since it [inherits](https://github.com/pylint-dev/astroid/blob/cdeb259e4258f7472c526cf373c35c2600f1c3c6/astroid/nodes/scoped_nodes/scoped_nodes.py#L1823C43-L1823C43) that method from [LocalsDictNodeNG](https://github.com/pylint-dev/astroid/blob/cdeb259e4258f7472c526cf373c35c2600f1c3c6/astroid/nodes/scoped_nodes/mixin.py#L139C7-L139C7).

Reply:
Okay! Can we then remove that code?

Reply:
I took a look into that just now. The template is used by [this function](https://github.com/pylint-dev/astroid/blob/main/astroid/brain/brain_typing.py#L187) which needs the `__args__` member since the function is inferring subscriptable types.
As an illustration, If I were to remove the code and use a similar ClassDef construction for the code above, this unitest fails:
`FAILED astroid/tests/brain/test_brain.py::TypingBrain::test_has_dunder_args - astroid.exceptions.InferenceError: StopIteration raised without any error information.`
In other words, running Pylint on this code:
```python
from typing import Union
NumericTypes = Union[int, float]
NumericTypes.__args__ #@
```
...gives: `E1101: Class 'Union' has no '__args__' member (no-member)`